dCommon/AMFDeserialize.cpp

```diff
@@ -0,0 +1,151 @@
+#include "AMFDeserialize.h"
+
+#include "AMFFormat.h"
+
+/**
+ * AMF3 Reference document https://rtmp.veriskope.com/pdf/amf3-file-format-spec.pdf
+ * I have only written the values that the LEGO Universe client sends to the server.  All others are left out.
+ * AMF3 Deserializer written by EmosewaMC
+ */
+AMFArrayValue* AMFDeserialize::Read(RakNet::BitStream* inStream, bool readStartOfArray, std::string parentKey) {
+    if (!inStream) return nullptr;
+    AMFArrayValue* values = new AMFArrayValue();
+    if (readStartOfArray) {
+        int8_t start;
+        int8_t size;
+        inStream->Read(start);
+        inStream->Read(size);
+    }
```

Comment:
The intent of this section is to read what would be in binary 09 01 from the start of an array.  This probably should be adjusted to follow the file spec a bit closer since this portion of any amf was generally always started with 09 01.

Reply:
FWIW, 0x09 is the array marker, which should not neer to be special cased. It's just that the root value can be any type.

Reply:
I hadn't thought about the fact that amf could be sent not contained within a base array.

Reply:
Yeah, I think it helps to compare it to JSON: `0`, `true`, `"hi"` are all valid json documents even though most APIs will produce something wrapped in `[...]` or `{...}`